### DISCUSSION TITLE
Multiple Plex Scrobbles

### DISCUSSION THREAD

**Discussion by glensc** [Taxel/PlexTraktSync#146] [Ideas]
_I tried to find existing issues on this topic, but could not find any._

Anyway, Plex does have data and PlexApi has methods to retrieve the data:

```python
                history = movie.history()
                if len(history) > 1:
                    print("%s Watching history (%d)" % (movie, len(history)))
                    for h in history:
                        print("- Watched: %s, %s" % (h.lastViewedAt, h.viewedAt))
```

```
<Movie:9772:Amélie> Watching history (2)
- Watched: 2021-02-07 20:57:44, 2021-02-07 20:57:44
- Watched: 2021-02-07 20:57:44, 2021-02-07 02:30:23
```

the problem with Plex is that you can't add a Plex watch with a timestamp. It will always be the current timestamp, and that messes up synchronization. But something is possible to achieve, maybe only one way sync of multiple watches.

**Comment by simonc56** [comment]
> the problem with Plex is that you can't add a Plex watch with a timestamp. It will always be the current timestamp

We know that, it's in readme.

> But something is possible to achieve, maybe only one way sync of multiple watches.

What do you mean ?

**Reply by glensc** [reply]
Like: take Plex watch history, and if those exact timestamps don't exist in Trakt, submit them to Trakt. But need to skip historical Plex watch entries, so like hardcode timestamp since when to start using this. or make it user configurable.

**Reply by glensc** [reply]
Ok, that timestamp_since solution still has problem if sync script has to take new trakt watch status record and set watched plex. this set watched will be with current timestamp, not trakt timestamp, so it will end up creating new "watched' in trakt.

Now, this could be avoided, to be able to detect who owns the Plex watch history item. The Movie object returned from `movie.history()` doesn't have any such property, but if you look at play history in Plex Web, you will see there columns like User/Player/Platorm. If we could set Player=PlexTraktSync, we could ignore all history entries with that Player.

**Reply by glensc** [reply]
When opening "View History", Plex Web makes request to `"/status/sessions/history/all?sort=viewedAt:desc&metadataItemID=9725"` which has `"accountID": 1`, and `"deviceID": 15` in response. 

Those are likely the User and Player.

**Reply by glensc** [reply]
The `deviceID` is present in XML but not in PlexAPI, looks like it needs a PR first:
- https://github.com/pkkid/python-plexapi/pull/704

**Reply by glensc** [reply]
Also, setting deviceId when making requests, is very tricky, need to hack internals of plex api:
- https://github.com/pkkid/python-plexapi/issues/706

**Comment by glensc** [comment]
Some more ideas: Could track some state in Plex labels
- https://www.reddit.com/r/PleX/comments/fgh793/python_plexapi_and_sharing_label/
- https://python-plexapi.readthedocs.io/en/latest/modules/media.html#plexapi.media.Label
- https://gist.github.com/JonnyWong16/4cfcf8ea50dab1b720b4d30e9a01835c

I haven't thought about the logic of how they could be useful, just posting some links for future reference
